### PR TITLE
Add lead form component

### PR DESCRIPTION
**Pull request by remidej**
Todo:

* Fix the styling of the lead form using Tailwind
* Add error and loading states to the lead form

### CODE REVIEW

**Comment by markkaylor** [comment]
Tailwind v2.0:
- IE 11 support dropped
- Requires Node.js 12.13.0 or higher in both your local and CI environments
- Had to follow this: [https://tailwindcss.com/docs/installation#post-css-7-compatibility-build](https://tailwindcss.com/docs/installation#post-css-7-compatibility-build)

File: frontend/components/elements/button.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import { buttonLinkPropTypes } from "utils/types";
+
+const Button = ({ button, appearance, compact = false, handleClick, loading = false }) => {
+  
+  const isLoading = () => {
+    if (loading) {
+      return (
+        <svg
```
Comment (alexandrebodin) [inline frontend/components/elements/button.js]: the loader deserves it's own compo file :)

File: frontend/components/sections/lead-form.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import Button from "../elements/button";
```
Comment (alexandrebodin) [inline frontend/components/sections/lead-form.js]: You could consider using a validator library such as yup js & even a form library (formik as it works great with yup) so it looks more like a really world app

Comment (soupette) [reply frontend/components/sections/lead-form.js]: Alex is right it is easier to use a validator library. Using formik will prevent you from managing the errors when the form is submitted.

However, if you choose to do it manually here is an approach based on what we do in the admin panel, using a reducer is easier but here's the example using a state:


```
const [{ modifiedData, formErrors }, setState] = useState({
    modifiedData: { email: '' },
    formErrors: {},
  });
```

File: frontend/components/sections/lead-form.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+import { useState } from 'react';
+import { fetchAPI } from 'utils/api';
+import * as yup from 'yup';
+import { Formik } from 'formik';
+
+const LeadForm = ({ data }) => {
+  const [loading, setLoading] = useState(false);
+
+  const schema = yup.object().shape({
+    email: yup.string().email(),
```
Comment (soupette) [inline frontend/components/sections/lead-form.js]: `email: yup.string().email().required()`

File: frontend/components/sections/lead-form.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            if (!valid) {
+              errors.email = 'Please enter a valid email';
+            }
+
+            return errors;
+          }}
+          onSubmit={async (values, { setSubmitting, setErrors }) => {
+            setLoading(true);
+            try {
+              await fetchAPI('/lead-form-submissions', {
```
Comment (soupette) [inline frontend/components/sections/lead-form.js:26]: You might need to remove the api errors here.

Comment (markkaylor) [reply frontend/components/sections/lead-form.js:26]: 👍  you mean something like ```setErrors({ api: null })``` ?

File: frontend/components/sections/lead-form.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    email: yup.string().email().required(),
+  })
+
+  return (
+    <div className="py-10 text-center">
+      <h1 className="text-3xl mb-10 font-bold mb-2">{data.title}</h1>
+      <div className="flex flex-col items-center flex-wrap ">
+        <Formik
+          initialValues={{ email: '' }}
+          validate={async (values) => {
```
Comment (soupette) [inline frontend/components/sections/lead-form.js]: With formik and yup you don't need to develop the validate function. You can directly use the [`validationSchema`](https://formik.org/docs/guides/validation#validationschema) prop.

**Review by soupette — COMMENTED** [review]
LGTM

**Comment by remidej** [comment]
LGTM

**Pull request merged**